pkg/datadog/datadog.go

```diff
@@ -73,21 +77,22 @@ func (ddman *DDMonitorManager) AddOrUpdate(monitor *datadog.Monitor) (*datadog.M
 		return provisioned, nil
 	}
 
+	merged := *monitor
```

Comment:
To recap my understanding here, currently:
* we create a monitor
* we see an event that may have caused a change to the monitor
* we retrieve the existing monitor via DD API
* we look at (a) the monitor we've constructed vs (b) the monitor we got from DD
* the monitor we got from DD probably has some extra fields that were filled in by the DD API (e.g. something like `CreationTime`), so DeepEqual always returns false

And you've fixed it by copying all non-zero fields from the monitor we get back from DD into the monitor we've created.

My only question is, would we ever want to update a field to a zero-value? With the new code, if we were trying to set `monitor.Message = ""`, IIUC it would get overwritten with the existing `Message`.

Reply:
It looks like it might just be the `ID` field that needs to get copied over

Reply:
Hmm you are correct! I can't think of a good reason to override a value with an empty value, but I can make some changes to the merge functionality so it doesn't touch the possible override values.

Reply:
Hmm you are correct! I can't think of a good reason to override a value with an empty value, but I can make some changes to the merge functionality so it doesn't touch the possible override values.

There are a few others, like the `Creator` struct and `OverallState`, as well as some things in the `Options` struct

Reply:
Updated code and broke out merging into it's own function.